src/transformers/tokenization_utils_fast.py

```diff
@@ -870,3 +878,58 @@ def train_new_from_iterator(
             kwargs["additional_special_tokens"] = additional_special_tokens
 
         return self.__class__(tokenizer_object=tokenizer, **kwargs)
+
+    def update_post_processor(self):
+        """
+        Updates the underlying post processor with the current `bos_token` and `eos_token`.
+        """
+        if not isinstance(self._tokenizer.post_processor, processors.TemplateProcessing) and not isinstance(
+            self._tokenizer.post_processor, processors.Sequence
+        ):
+            return
+
+        warnings.warn(
+            "Warning overwriting the original postProcessor in order to update `bos_token` or `eos_token`. "
+            "Reload the tokenizer without these parameters if that is not desired"
+        )
+
+        bos = self.bos_token
+        bos_token_id = self.bos_token_id
+        if bos is None and self.add_bos_token:
+            raise ValueError("add_bos_token = True but bos_token = None")
+
+        eos = self.eos_token
+        eos_token_id = self.eos_token_id
+        if eos is None and self.add_eos_token:
+            raise ValueError("add_eos_token = True but eos_token = None")
+
+        single = f"{(bos + ':0 ') if self.add_bos_token else ''}$A:0{(' ' + eos + ':0') if self.add_eos_token else ''}"
+        pair = f"{single}{(' ' + bos + ':1') if self.add_bos_token else ''} $B:1{(' ' + eos + ':1') if self.add_eos_token else ''}"
+
+        special_tokens = []
+        if self.add_bos_token:
+            special_tokens.append((bos, bos_token_id))
+        if self.add_eos_token:
+            special_tokens.append((eos, eos_token_id))
+        if special_tokens:
+            self._tokenizer.post_processor = processors.TemplateProcessing(
+                single=single, pair=pair, special_tokens=special_tokens
+            )
+
+    @property
+    def add_eos_token(self):
+        return self._add_eos_token
+
+    @property
+    def add_bos_token(self):
+        return self._add_bos_token
+
+    @add_eos_token.setter
+    def add_eos_token(self, value):
+        self._add_eos_token = value
+        self.update_post_processor()
+
+    @add_bos_token.setter
+    def add_bos_token(self, value):
+        self._add_bos_token = value
+        self.update_post_processor()
```

Comment:
Cool! Let's document add_eos and add_bos

Reply:
sounds good! should it only be an autodoc? like in gemma [here](https://huggingface.co/docs/transformers/v4.44.0/en/model_doc/gemma#transformers.GemmaTokenizer:~:text=for%20BPE%2Ddropout.-,add_bos_token%20(bool%2C%20optional%2C%20defaults%20to%20True)%20%E2%80%94%20Whether%20or%20not%20to%20add%20an%20bos_token%20at%20the%20start%20of%20sequences.,-add_eos_token%20(bool)

Reply:
Yeah, or in the __init__ 's doc (of kwargs)

Reply:
@ArthurZucker sorry I'm not sure what is the init doc of kwargs?